src/service/viewer-impl.js

```diff
@@ -802,6 +802,25 @@ export class Viewer {
   }
 
   /**
+   * Get the fragment from the url or the viewer.
+   * Strip leading '#' in the fragment
+   * @return {!Promise<string|undefined>}
+   */
+  getFragment() {
+    if (!this.isEmbedded_) {
```

Comment:
In non-embedded case, this method just return the url hash.
In embedded case, I think the search viewer cannot set the share-tracking fragment in the url because the search viewer doesn't know the share-tracking fragment when first generating the amp url. For example the search viewer url: https://www.google.com/amp/s/www.foobar.com/some-article#.1a2b3c4d. But the amp url generated before runtime would be https://www.foobar.com/some-article?amp_js_v=5#origin=... without the share-tracking info. At runtime, the .1a2b3c4d info is passed to search viewer, then search viewer send a message to amp page.